merlin/systems/dag/dictarray.py

```diff
@@ -13,44 +13,149 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 #
+from enum import Enum
 from typing import Dict, Optional
 
 import numpy as np
 
-from merlin.core.protocols import SeriesLike, Transformable
+from merlin.core.dispatch import HAS_GPU
+from merlin.core.protocols import SeriesLike
+
+if HAS_GPU:
```

Comment:
`HAS_GPU` from the dispatch module also requires cudf and dask_cudf are installed. Either replacing this with a try/except ImportError or other method could allow us to run this without these other restrictions.